### ISSUE TITLE
JSONB output to terminal, csv etc uses wrong UTF8 escaping, does not match result of cast to string

### ISSUE THREAD

**Issue by arctica** [cockroachdb/cockroach#86917]
**Describe the problem**

The string output of a JSONB value when printed to the terminal or for example csv encodes unicode characters via \x escape sequence. This does not happen when casting the JSONB first to a string. JSON does not consider \x escapes valid.

This causes rows which contain JSONB values with these characters exported to csv to not be possible to import via for example IMPORT INTO as JSON decoding fails.

**To Reproduce**

```
> SELECT '"Ä"'::jsonb, '"Ä"'::jsonb::string;
    jsonb    | text
-------------+-------
  "\xc3\x84" | "Ä"
(1 row)
```

1. cockroach sql -e 'SELECT * FROM t' --format=csv > t.csv
2. IMPORT INTO t2 CSV DATA ....
3. ERROR: could not parse JSON: unable to decode JSON: invalid character 'x' in string escape code

**Expected behavior**

JSONB columns should encode to the same when output to terminal, csv etc as if they were cast to string first.

**Environment:**

 - CockroachDB version: v22.1.6
 - Server OS: Linux
 - Client app: cockroach sql

Jira issue: CRDB-19036

**Comment by rytaft** [comment]
This seems to be fixed on master. I bisected to this commit:
```
commit fcc7de4c4864f4763cd8d496150c4fd88a183339
Author: Rafi Shamim <rafi@cockroachlabs.com>
Date:   Wed Apr 27 10:03:14 2022 -0400

    cli: switch to jackc/pgx
    
    - Address TODO about RowsAffected. Unlike lib/pq pgx doesn't use
      incorrect command tags for queries that return 0 rows, so we don't
      need this special logic.
    - BYTEA values are now formatted correctly. Along with this, we need
      some special handling for formatting NAME[], "char", TIME, INTERVAL,
      NUMERIC, and UUID types.
    - We needed an additional "multi result set" rows object. lib/pq would
      handle this with one interface, but this isn't possible using
      pgx/pgconn.
    
    Release note (cli change): BYTEA values are now formatted according to
    the bytea_output session setting.
    
    Release note (cli change): The statement tag displayed for INSERT
    statements now has the full information returned by the server. It is
    the string "INSERT" followed by the OID of the row that was inserted
    (which is currently always 0 in CockroachDB), followed by the number of
    rows inserted.
    
    Release note (backward-incompatible change): In the cockroach CLI, BOOLEAN
    values are now formatted as `t` or `f` instead of `True` or `False`.
```
cc @rafiss not sure if any of this is backportable.

**Comment by rafiss** [comment]
No, that change isn't able to be back ported. as a workaround, you can try using a different cli client like `psql` to get the CSV output.

**Comment by rafiss** [comment]
This will be fixed in v22.2 -- there are alpha releases available here https://www.cockroachlabs.com/docs/releases/index.html#testing-releases and a production-ready release will be out in the coming months.
